app/src/main/java/com/doctor/yumyum/data/remote/api/UserApi.kt

```diff
@@ -0,0 +1,30 @@
+package com.doctor.yumyum.data.remote.api
+
+import com.doctor.yumyum.data.model.NicknamePatchModel
+import com.doctor.yumyum.data.model.SignUpModel
+import com.doctor.yumyum.data.remote.response.GetNicknameResponse
+import okhttp3.ResponseBody
+import retrofit2.Response
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.PATCH
+import retrofit2.http.Query
+
+interface NicknameCreationService {
+    @GET("/api/v1/users/nickname")
+    suspend fun getNicknameApi(): Response<GetNicknameResponse>
+}
+
+interface NicknameValidationService {
+    @GET("/api/v1/users/me/nickname/validation")
+    suspend fun validateNicknameApi(
+        @Query("nickname") nickname: String
+    ):Response<ResponseBody>
+}
+
+interface NicknameService {
+    @PATCH("/api/v1/users/me/nickname")
+    suspend fun patchNicknameApi(
+        @Body nicknamePatchModel: NicknamePatchModel
+    ):Response<ResponseBody>
+}
```

Comment:
하나의 서비스 안에 여러 api들을 같이 정의해도 될 것 같습니다! 그리고 함수명에는 Api 안붙이고 어떤 기능인지만 적는 게 좋을 것 같아용
```suggestion
interface NicknameService {
    @GET("/api/v1/users/nickname")
    suspend fun getNickname(): Response<GetNicknameResponse>
    
    @GET("/api/v1/users/me/nickname/validation")
    suspend fun validateNickname(
        @Query("nickname") nickname: String
    ):Response<ResponseBody>

    @PATCH("/api/v1/users/me/nickname")
    suspend fun patchNickname(
        @Body nicknamePatchModel: NicknamePatchModel
    ):Response<ResponseBody>
}
```